pkg/operator/credentialsrequest/credentialsrequest_controller_gcp_test.go

```diff
@@ -332,6 +333,50 @@ func TestCredentialsRequestGCPReconcile(t *testing.T) {
 				assert.True(t, cr.Status.Provisioned)
 			},
 		},
+		{
+			name: "cred minted and up to date and secret exist without root creds",
+			existing: []runtime.Object{
+				testOperatorConfig(""),
+				createTestNamespace(testNamespace),
+				testClusterVersion(),
+				testInfrastructure(testInfraName),
+
+				// already minted, last synced 2 hours ago
+				func() *minterv1.CredentialsRequest {
+					cr := testGCPCredentialsRequest(t)
+					cr.Status.Provisioned = true
+					cr.Status.LastSyncTimestamp = &metav1.Time{Time: time.Now().Add(-2 * time.Hour)}
+
+					return cr
+				}(),
+
+				// target secret exists
+				createTestNamespace(testSecretNamespace),
+				testGCPCredsSecret(testSecretNamespace, testSecretName, `{"private_key_id": "fakeServiceAccountID"}`),
+
+				// only the read-only creds exist
+				testGCPCredsSecret("kube-system", constants.GCPCloudCredSecretName, "READ ONLY SERVICE ACCOUNT DATA"),
+			},
+			mockReadGCPClient: func(mockCtrl *gomock.Controller) *mockgcp.MockClient {
+				mockGCPClient := mockgcp.NewMockClient(mockCtrl)
+
+				// needs update
+				mockGetRole(mockGCPClient)
+				mockListServicesEnabled(mockGCPClient)
+				mockGetProjectName(mockGCPClient)
+				mockListServiceAccountKeys(mockGCPClient, "fakeServiceAccountID")
+				mockGetServiceAccount(mockGCPClient)
+				mockGetProjectIamPolicy(mockGCPClient, testValidPolicyBindings)
+
+				return mockGCPClient
+			},
+			validate: func(c client.Client, t *testing.T) {
+				cr := getCredRequest(c)
+
+				lastSynced := cr.Status.LastSyncTimestamp.Time
+				assert.WithinDuration(t, time.Now(), lastSynced, time.Second*5, "expected a recent last synced status")
+			},
+		},
```

Comment:
Could we get tests for: cred needs a modification but no root cred, cred is missing entirely but no root cred.